kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JdbcDatabaseDialect.scala

```diff
@@ -19,14 +19,71 @@ package org.apache.kyuubi.server.metadata.jdbc
 
 trait JdbcDatabaseDialect {
   def limitClause(limit: Int, offset: Int): String
+  def insertOrReplace(
+      table: String,
+      cols: Seq[String],
+      keyCol: String,
+      keyVal: String): String
 }
 
 class GenericDatabaseDialect extends JdbcDatabaseDialect {
   override def limitClause(limit: Int, offset: Int): String = {
     s"LIMIT $limit OFFSET $offset"
   }
+
+  override def insertOrReplace(
+      table: String,
+      cols: Seq[String],
+      keyCol: String,
+      keyVal: String): String = {
+    s"""
+       |INSERT INTO $table (${cols.mkString(",")})
+       |SELECT ${cols.map(_ => "?").mkString(",")}
+       |WHERE NOT EXISTS (
+       |  SELECT 1 FROM $table WHERE $keyCol = '$keyVal')
+       |)
+       |""".stripMargin
```

Comment:
cc @pan3793 how about now?

I have to inject the keyVal here.

Reply:
lgtm